uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsenceType.php

```diff
@@ -540,4 +540,39 @@ public static function getOneWithMustTakePublicHolidayAsLeaveRequest() {
 
     return null;
   }
+
+  /**
+   * Returns the date the TOIL will expire based on the date given to the method.
+   *
+   * @param int $typeId
+   *   The Absence type ID
+   * @param string $date
+   *   The date to calculate TOIL expiry date
+   *
+   * @return null|false|string
+   */
+  public static function calculateToilExpiryDate($typeId, $date) {
+    $absenceType = new self();
+    $absenceType->id = $typeId;
+    $absenceType->find(true);
+
+    if (!$absenceType->allow_accruals_request) {
+      return null;
+    }
+    if (!$absenceType->accrual_expiration_unit || !$absenceType->accrual_expiration_duration){
```

Comment:
What of changing to a variable instead of an extra function? 
```
$toilNeverExpires = !$this->accrual_expiration_unit || !$this->accrual_expiration_duration
if($this->toilNeverExpires) {
  return false;
}
```